spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/config/AbstractJdbcConfiguration.java

```diff
@@ -96,8 +101,22 @@ public JdbcConverter jdbcConverter(JdbcMappingContext mappingContext, NamedParam
 	 * @return will never be {@literal null}.
 	 */
 	@Bean
-	public JdbcCustomConversions jdbcCustomConversions() {
-		return new JdbcCustomConversions();
+	public JdbcCustomConversions jdbcCustomConversions(Dialect dialect) {
```

Comment:
A potential way out could be injecting `ApplicationContext` into this class making it `ApplicationContextAware` and handle the lookup through `context.getBean(…)`.